config/configmapprovider/file.go

```diff
@@ -46,6 +46,21 @@ func (fmp *fileMapProvider) Retrieve(_ context.Context, _ func(*ChangeEvent)) (R
 	return &simpleRetrieved{confMap: cp}, nil
 }
 
+func (fmp *fileMapProvider) RetrieveValue(_ context.Context, _ func(*ChangeEvent), selector string, paramsConfigMap *config.Map) (RetrievedValue, error) {
+	if selector == "" {
+		return nil, errors.New("config file not specified")
+	}
+
+	// TODO: support multiple file paths in selector (use some sort of delimiter).
+
+	cp, err := config.NewMapFromFile(selector)
+	if err != nil {
+		return nil, fmt.Errorf("error loading config file %q: %w", fmp.fileName, err)
+	}
+
+	return &simpleRetrievedValue{value: cp}, nil
+}
```

Comment:
Very confusing. RetrieveValue seems to be a "subset" of the RetrieveMap from the description but here is a super set.

Also you cannot return any "random" instance (interface{}) because we have to unmarshal after.

Reply:
>Very confusing. RetrieveValue seems to be a "subset" of the RetrieveMap from the description but here is a super set.

It is a superset. Description needs to be fixed.

>Also you cannot return any "random" instance (interface{}) because we have to unmarshal after.

It is in the TODO to fix this.

Reply:
Improved the description a bit. It still all needs a good pass of clarification, it is still draft.